library/common/extensions/filters/http/assertion/filter.proto

```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
+
+package envoymobile.extensions.filters.http.assertion;
+
+import "envoy/config/tap/v3/common.proto";
+
+import "validate/validate.proto";
+
+message Assertion {
+  // The match configuration. If the configuration matches the request frames the filter will send
+  // a local reply with Http::Code::OK on the last frame of the request stream (continuing on
+  // intervening frames). Otherwise, it will send a local reply with Http::Code::BadRequest.
+  envoy.config.tap.v3.MatchPredicate match_config = 1 [(validate.rules).message = {required: true}];
+}
```

Comment:
```suggestion
  // The match configuration. If the configuration matches the request frames, the filter will send
  // a local reply with Http::Code::OK on the last frame of the request stream (continuing on
  // intervening frames). Otherwise, it will send a local reply with Http::Code::BadRequest.
  envoy.config.tap.v3.MatchPredicate match_config = 1 [(validate.rules).message = {required: true}];
}
```
It'd be nice if we could validate here that there's at least 1 matcher rule within the config (since it's asserted above), but not a big deal. We should probably document it here though

Reply:
Right, that would need a change to the upstream proto.